controllers/authentication.ts

```diff
@@ -0,0 +1,66 @@
+import {Request, Response, NextFunction} from "express";
+import mongoose from "mongoose";
+const {User} = require("../models/user");
+const bcrypt = require("bcrypt");
+const mailgun = require("mailgun-js");
+const jwt = require("jsonwebtoken");
+const mg = mailgun({
+   apiKey: process.env.MG_API_KEY,
+   domain: process.env.MG_DOMAIN,
+});
+
+exports.send = (name: string, email: string, password: string) => {
+   const token = jwt.sign({name, email, password}, process.env.JWT_KEY, {
+      expiresIn: "15m",
+   });
+
+   mg.messages().send(
+      {
+         from: "noreply@camp.project.com",
+         to: "camp.project.aj@gmail.com", //later change to <email>
+         subject: "Email confirmation",
+         html: `
+         <h2>PLease click the link below to confirm your email</h2>
+         <p>localhost:3000/users/auth/${token}</p>
+      `,
+      },
+      (err, body) => {
+         if (err) return err;
+      }
+   );
+};
+
+exports.activateAccount = (req: Request, res: Response, next: NextFunction) => {
+   const token = req.body.token;
+
+   if (token) {
```

Comment:
Im mniej kodu jest w zagnieżdżeniu, tym kod jest czytelniejszy. Można spróbować robić to w myśl takiego sposobu:
```
if(!token){
    return res.status(500).json({err: "lack of token!?"});
}
```
A potem cała reszta idzie normalnie - już bez zagnieżdżania :)